packages/ag-charts-community/src/scene/layer.ts

```diff
@@ -125,16 +105,95 @@ export class Layer extends Group {
             ctx.restore();
         }
 
+        ctx.restore();
+        ctx.verifyDepthZero?.(); // Check for save/restore depth of zero!
+
+        return true;
+    }
+
+    private renderNext(renderCtx: RenderContext, forceRender: boolean | 'dirtyTransform') {
+        if (!this.getVisibility()) {
+            return true;
+        } else if (!this.layer) {
+            super.render(renderCtx);
+            return false;
+        }
+
+        const { isDirty, isChildDirty, isChildLayerDirty } = this.isDirty(renderCtx);
+
+        if (forceRender) {
+            this.layer.clear();
+        }
+
+        if (isDirty || isChildDirty || isChildLayerDirty || forceRender) {
+            const layerRenderCtx: RenderContext = {
+                ...renderCtx,
+                ctx: this.layer.context,
+                forceRender,
+            };
+
+            super.render(layerRenderCtx, false);
+
+            this.layer.context.verifyDepthZero?.(); // Check for save/restore depth of zero!
+        } else {
+            this.debugSkip(renderCtx);
+            this.markClean({ recursive: false });
+        }
+
+        const { ctx } = renderCtx;
+        ctx.save();
+        ctx.resetTransform();
+        ctx.globalAlpha = this.getComputedOpacity();
+        this.layer.drawImage(ctx);
+        ctx.restore();
+
+        return true;
+    }
+
+    override render(renderCtx: RenderContext) {
```

Comment:
Nit: Maybe move this before the new methods above to better preserve line history? (Might need to rebase/squash to achieve this)

Reply:
Will do this after you merge your z-index branch - otherwise I'll get conflicts